ftw/upgrade/tests/base.py

```diff
@@ -50,6 +53,65 @@ def grant(self, *roles):
         setRoles(self.portal, TEST_USER_ID, list(roles))
         transaction.commit()
 
+    def login(self, user, browser=None):
+        """Login a user by setting the security manager.
+
+        The login method logs in a user by creating a new security manager
+        with this user.
+        The ``user`` may be a userid or a user object.
+
+        >>> self.login(acl_users.getUserById('john.doe'))
+        >>> self.login('john.doe')
+
+        The optional ``browser`` keyword argument allows to pass in an
+        ``ftw.testbrowser`` browser instance, which will be logged in with
+        the same user.
+
+        >>> self.login(user_object, browser=browser)
+
+        When the method is used as context manager, the prior security manager
+        will be restored on exit:
+
+        >>> with self.login(admin):
+        ...     do_something_as_admin()
+
+        :param user: A user object or a user name.
+        :type user: string or user object
+        :param browser: A ``ftw.testbrowser`` instance to log in too.
```

Comment:
`login to` or `log in to` or `log into` - but def not `too`.  I prefer `log into` (as does Facebook)

https://english.stackexchange.com/questions/5302/log-in-to-or-log-into-or-login-to

Reply:
😂